### PR TITLE
Fix: Added github id with github user details

### LINKED ISSUE #574: Use Github IDs instead of names in DB
Currently user, repository, organization are all stored by their name in the database. User, repo or organization can change their name and I guess this is not that common but I still think it would be better to store IDs that we get from the Github API instead of names. 

---

- [ ] Store and use user ID. 
- [ ] Store and save username. 
- [ ] Ensure that user name is updated every time user ID is accessed, in case a username is updated.

### PR DESCRIPTION
**Pull request by baibhavKumar1**

### CODE REVIEW

File: src/adapters/supabase/helpers/client.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export type UserProfileAdditions = {
 };
 const getDbDataFromUserProfile = (userProfile: UserProfile, additions?: UserProfileAdditions) => {
   return {
+    user_id: userProfile.id,
```
Comment (whilefoo) [inline src/adapters/supabase/helpers/client.ts:93]: you need to also add this field to `upsertUser`

**Review by whilefoo — CHANGES_REQUESTED** [review]
You also need to do this for other tables `access`, `penalty`, `wallets`, `multiplier`

File: src/adapters/supabase/types/database.ts [hunk 2]
```diff
@@ -145,6 +145,7 @@ export interface Database {
           user_login: string;
           user_name: string;
           user_type: string | null;
+          user_id: number;
```
[the remark this answers is not in the document]
Comment (Wholesomebruh) [reply src/adapters/supabase/types/database.ts:148]: This was auto-generated.

File: supabase/migrations/20230111145014_initial_schema.sql [hunk 3]
```diff
@@ -6,6 +6,7 @@ CREATE TYPE issue_status AS ENUM (
 );
 
 CREATE TABLE users (
+    user_id bigint NOT NULL,
```
[the remark this answers is not in the document]
Comment (Wholesomebruh) [reply supabase/migrations/20230111145014_initial_schema.sql:9]: I'm unclear about this. Please describe.

File: src/adapters/supabase/helpers/client.ts (hunk 1 above)
Comment (Wholesomebruh) [reply src/adapters/supabase/helpers/client.ts:93]: Done, will be implemented in next commit.

File: supabase/migrations/20230111145014_initial_schema.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (whilefoo) [reply supabase/migrations/20230111145014_initial_schema.sql:9]: Migrations are only run once, so when you want to make a change to the database you need to make a new migration that will alter the database.

**Review by web4er — CHANGES_REQUESTED** [review]
A QA to show IDs being inserted correctly without error will be helpful here. One way to do so will be to run the commands when the ID is supposed to be inserted and attach a screenshot. Example https://github.com/web4erOrg/ubiquibot/issues/46#issuecomment-1680271472

**Pull request closed without merging**